gammapy/spectrum/make.py

```diff
@@ -0,0 +1,220 @@
+        return WcsGeom.create(
+            skydir=self.region.center,
+            width=self.width,
+            binsz=self.binsz,
+            proj="TAN",
+            coordsys=coordsys,
+        )
+
+    @lazyproperty
+    # TODO: move this to a RegionGeom class
+    def region_solid_angle(self):
+        """Solid angle of the region"""
+        geom = self.geom_ref
+        coords = geom.get_coord()
+        solid_angle = geom.solid_angle()
+        mask = self.region.contains(coords.skycoord, wcs=geom.wcs)
+        return solid_angle[mask].sum()
+
+    def make_counts(self, observation):
+        """Make counts
+
+        Parameters
+        ----------
+        observation: `DataStoreObservation`
+            Observation to compute effective area for.
+
+        Returns
+        -------
+        counts : `CountsSpectrum`
+            Counts spectrum
+        """
+        energy_hi = self.e_reco[1:]
+        energy_lo = self.e_reco[:-1]
+
+        counts = CountsSpectrum(energy_hi=energy_hi, energy_lo=energy_lo)
+        events_region = observation.events.select_region(
+            self.region, wcs=self.geom_ref.wcs
+        )
+        counts.fill(events_region)
+        return counts
+
+    def make_background(self, observation):
+        """Make background
+
+        Parameters
+        ----------
+        observation: `DataStoreObservation`
+            Observation to compute effective area for.
+
+        Returns
+        -------
+        background : `CountsSpectrum`
+            Background spectrum
+        """
+        offset = observation.pointing_radec.separation(self.region.center)
+        energy_hi = self.e_reco[1:]
+        energy_lo = self.e_reco[:-1]
+
+        bkg = observation.bkg
+
+        data = bkg.evaluate_integrate(
+            fov_lon=0 * u.deg, fov_lat=offset, energy_reco=self.e_reco
+        )
+
+        data *= self.region_solid_angle
+        data *= observation.observation_time_duration
+
+        counts = CountsSpectrum(
+            energy_hi=energy_hi, energy_lo=energy_lo, data=data.to_value(""), unit=""
+        )
+        return counts
+
+    def make_aeff(self, observation):
+        """Make effective area
+
+        Parameters
+        ----------
+        observation: `DataStoreObservation`
+            Observation to compute effective area for.
+
+        Returns
+        -------
+        aeff : `EffectiveAreaTable`
+            Effective area table.
+        """
+        offset = observation.pointing_radec.separation(self.region.center)
+        aeff = observation.aeff.to_effective_area_table(offset, energy=self.e_true)
+
+        if self.containment_correction:
+            if not isinstance(self.region, CircleSkyRegion):
+                raise TypeError(
+                    "Containment correction only support for circular regions."
+                )
+            psf = observation.psf
+
+            if isinstance(psf, EnergyDependentMultiGaussPSF):
```

Comment:
Actually I realised this call is not necessary at all, because all our current psf irf classes support the `.to_energy_dependent_table_psf()` method. I just removed it...